### ISSUE TITLE
docker connection rather than localhost?

### ISSUE THREAD

**Issue by ryantupo** [bee-queue/bee-queue#424]
using this ->

const Queue = require('bee-queue', {
  prefix: 'bq',
  stallInterval: 5000,
  nearTermWindow: 1200000,
  delayedDebounce: 1000,
  redis: {
    host: 'redis',
    port: 6379,
    db: 0,
    options: {},
  },
  isWorker: true,
  getEvents: true,
  sendEvents: true,
  storeJobs: true,
  ensureScripts: true,
  activateDelayedJobs: false,
  removeOnSuccess: false,
  removeOnFailure: false,
  redisScanCount: 100,
});

my docker-compose file ->

version: '3.1'

volumes:
  mysql:
  node:

services:
    mysql:
        container_name: mysql
        image: mysql:latest
        command: --default-authentication-plugin=mysql_native_password
        restart: always
        env_file: ./.devcontainer/devcontainer.env
        cap_add:
          - SYS_NICE
        ports:
            - 3306:3306
        volumes:
          - mysql:/var/lib/mysql:delegated

    redis:
        container_name: redis
        image: redis:latest
        restart: always
        ports:
          - 6379:6379

    node:
      container_name: node
      build:
        context: .
        dockerfile: ./etc/docker/node/Dockerfile
      env_file: ./.devcontainer/devcontainer.env
      ports:
        - 80:8641
      links:
        - mysql
        - redis
      volumes:
        - .:/workspace:cached
        - node:/workspace/node_modules:delegated

my link is "redis"

and getting this in the file im using queuebee ->>

events.js:291
      throw er; // Unhandled 'error' event
      ^

Error: Redis connection to 127.0.0.1:6379 failed - connect ECONNREFUSED 127.0.0.1:6379
    at TCPConnectWrap.afterConnect [as oncomplete] (net.js:1144:16)
Emitted 'error' event on Queue instance at:
    at /workspace/node_modules/bee-queue/lib/queue.js:30:40
    at processTicksAndRejections (internal/process/task_queues.js:79:11) {
  errno: 'ECONNREFUSED',
  code: 'ECONNREFUSED',
  syscall: 'connect',
  address: '127.0.0.1',
  port: 6379
}

**Comment by ryantupo** [comment]
keeps trying to connect to localhost...

**Comment by add1ct3dd** [comment]
What was the solution to this? I'm also getting it where it's ignoring the redisHost I'm putting in...

**Comment by szcrow** [comment]
I'm running into the same issue. Running `console.log(queue.settings.redis)` it correctly prints out:

```js
connector: StandaloneConnector {
     connecting: true,
     disconnectTimeout: 2000,
     options: {
     host: 'redis',  // correct docker host
     port: 6379,
     ...
     }
}
```

however, when trying to process the queue (the next line of code), it errors out with:
```
Error: connect ECONNREFUSED 127.0.0.1:6379
    at TCPConnectWrap.afterConnect [as oncomplete] (node:net:1495:16)
Emitted 'error' event on Queue instance at:
    at /app/node_modules/bee-queue/lib/queue.js:30:40
    at process.processTicksAndRejections (node:internal/process/task_queues:77:11) {
 errno: -111,
 code: 'ECONNREFUSED',
  syscall: 'connect',
  address: '127.0.0.1',  // the wrong address!
  port: 6379
}
```

where is `127.0.0.1` set?

**Comment by szcrow** [comment]
@compwright 

that is not the issue. i am not using links (or depends_on for that matter). just standard docker networking.

in any case it does not make any sense that it resolves the hostname 'redis' to '127.0.0.1'.
